engine/lite/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/TestRunner.scala

```diff
@@ -48,11 +50,27 @@ class InterpreterTestRunner[F[_] : InterpreterShape : CapabilityTransformer : Ef
 
     //TODO: probably we don't need statics here, we don't serialize stuff like in Flink
     val collectingListener = ResultsCollectingListenerHolder.registerRun(variableEncoder)
-
     //in tests we don't send metrics anywhere
     val testContext = LiteEngineRuntimeContextPreparer.noOp.prepare(testJobData(process))
     val componentUseCase: ComponentUseCase = ComponentUseCase.TestRuntime
 
+    val expressionEvaluator: (Expression, Parameter, Context, NodeId) => Validated[NonEmptyList[PartSubGraphCompilationError], (String, AnyRef)] = {
+      val validationContext = GlobalVariablesPreparer(modelData.processWithObjectsDefinition.expressionConfig).emptyValidationContext(process.metaData)
+      val evaluator = ExpressionEvaluator.unOptimizedEvaluator(modelData)
+      val expressionCompiler = ExpressionCompiler.withoutOptimization(modelData).withExpressionParsers {
+        case spel: SpelExpressionParser => spel.typingDictLabels
+      }
+      (expression: Expression, parameter: Parameter, context: Context, nodeId: NodeId) => {
+        expressionCompiler
+          .compile(expression, Some(parameter.name), validationContext, parameter.typ)(nodeId)
+          .map(typedExpression => {
+            val param = evaluatedparam.Parameter(typedExpression, parameter)
+            parameter.name -> evaluator.evaluateParameter(param, context)(nodeId, process.metaData).value
```

Comment:
Please return only the second element of the tuple. You already have the parameter (name) in place of the invocation. Can you replace:
```
invocationOf(lambdaParam => { 
.... 
})
```
with more concise version that we prefer in our code:
```
invocationOf { lambdaParam =>
...
}
```
here and in other places?

Reply:
Thanks, I've moved param name from tuple to different place, I've also changed invocations style as suggested :+1: